lib/engine-jit/src/artifact.rs

```diff
@@ -141,20 +137,26 @@ impl JITArtifact {
     }
 
     /// Deserialize a JITArtifact
-    pub fn deserialize(jit: &JITEngine, bytes: &[u8]) -> Result<Self, DeserializeError> {
+    pub unsafe fn deserialize(jit: &JITEngine, bytes: &[u8]) -> Result<Self, DeserializeError> {
         if !Self::is_deserializable(bytes) {
             return Err(DeserializeError::Incompatible(
                 "The provided bytes are not wasmer-jit".to_string(),
             ));
         }
 
-        let inner_bytes = &bytes[Self::MAGIC_HEADER.len()..];
+        let mut inner_bytes = &bytes[Self::MAGIC_HEADER.len()..];
+
+        let metadata_len = leb128::read::unsigned(&mut inner_bytes).map_err(|_e| {
+            DeserializeError::CorruptedBinary("Can't read metadata size".to_string())
+        })?;
+        let metadata_slice: &'static [u8] =
+            std::slice::from_raw_parts(&inner_bytes[12] as *const u8, metadata_len as usize);
```

Comment:
How is this safe code? Why isn't the pointer invalidated / dangling?

Reply:
Oh, the function is `unsafe`. It's not clear to me that `'static` lifetime is correct for this reference. Maybe @MarkMcCaskey can take a look?

Reply:
100% agreed, `'static` makes me real concerned, left feedback in my comments below